app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -1,43 +1,85 @@
 package org.oppia.app.parser
 
 import org.oppia.app.model.Fraction
+import org.oppia.app.topic.FractionParsingErrors
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d)+$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d)+ ?(\d+) ?/ ?(\d)+$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  private val fractionRegex = """^\s*-?\s*((\d*\s*\d+\s*\/\s*\d+)|\d+)\s*$""".toRegex()
+
+  fun fromRawInputString(inputText: String): FractionParsingErrors {
+    var denominator = 1
+    var rawInput: String = inputText.normalizeWhitespace()
+    if (!inputText.matches(invalidCharsRegex))
+      return FractionParsingErrors.INVALID_CHARS
+    if (!fractionRegex.matches(inputText)) {
+      return FractionParsingErrors.INVALID_FORMAT
+    }
+    rawInput = rawInput.trim();
+    if (rawInput.startsWith("-")) {
+      // Remove the negative char from the string.
+      rawInput = rawInput.substring(1).trim()
+    }
+    // Filter result from split to remove empty strings.
+    var numbers = Pattern.compile("[\\s|/]+").split(rawInput)
+    if (numbers.size == 1) {
+    } else if (numbers.size == 2) {
+      denominator = parseInt(numbers[1])
+    } else {
+      denominator = parseInt(numbers[2])
+    }
+    if (denominator == 0) {
+      return FractionParsingErrors.DIVISION_BY_ZERO
+    }
+    return FractionParsingErrors.VALID
+  }
 
   fun getFractionFromString(text: String): Fraction {
+    //for testing the validation in a single method
     // Normalize whitespace to ensure that answer follows a simpler subset of possible patterns.
     val inputText: String = text.normalizeWhitespace()
-    return parseMixedNumber(inputText)
-      ?: parseFraction(inputText)
-      ?: parseWholeNumber(inputText)
-      ?: throw IllegalArgumentException("Incorrectly formatted fraction: $text")
+    if (inputText.matches(invalidCharsRegex))
+      return parseMixedNumber(inputText)
+        ?: parseFraction(inputText)
+        ?: parseWholeNumber(inputText)
+        ?: throw IllegalArgumentException(FractionParsingErrors.INVALID_FORMAT.getError())
```

Comment:
I don't think we need this additional text error handling here or below. We really only need the parsing errors when checking for correctness, not when performing the actual parsing. That ought to fail with a crash, or return a default Fraction.

Reply:
Reverted